### PR TITLE
Feature sb 175 new header

### PR DESCRIPTION
**Pull request by VladManukian**
### **_Changes:_**
1. Refactor wrong structure
2. Added new icon for header
3. Added badge for notification
4. Header are separate for buyer/vendor/unregister users

### **_Task: [Jira](https://team-jfqe9wlknaql.atlassian.net/browse/SB-175)_**

unregister

HEADER BUYER
HEADER VENDOR

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/Header/guest-user-content/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+						{t("header.buttons.LogIn")}
+					</StyledButtonLogIn>
+					<StyledButtonSignUp href={AppRoute.SIGN_UP}>
+						{t("header.buttons.SignUp")}
+					</StyledButtonSignUp>
+				</ButtonsContainer>
+				<StyledBadge
+					// Pass here you product request count or order count
+					badgeContent={2}
+				>
```
Comment (likeri29) [inline src/components/Header/guest-user-content/index.tsx]: Is this a comment for further adjustments or for what?

Comment (lyudad) [reply src/components/Header/guest-user-content/index.tsx]: in this case better to add //TODO Pass here you product request count or order count
its will be to easy to check all comments that need be replaced, by word TODO

File: src/components/Header/main-logo/index.tsx [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
-import { CircularElement } from "~/components/Footer/Circular.tsx";
 import { LogoContainer, StyledTypography } from "~/components/Footer/styles.ts";
-import { Link } from "~/components/Header/Link.tsx";
-import { colors } from "~/libs/constants/color.ts";
+import { CircularElement } from "~/components/circular/index.tsx";
 import { AppRoute } from "~/libs/constants/index.ts";
+import theme from "~/theme.ts";
 
-export const MainLogo: React.FC = () => {
+import { Link } from "../../link/index.tsx";
```
Comment (lyudad) [inline src/components/Header/main-logo/index.tsx]: alias

File: src/components/link/index.tsx
Comment (lyudad) [inline src/components/link/index.tsx:1]: why "links" in lower case, but Header in upper case, they are in the same level

File: src/components/common/wrappers/footer-wrapper.tsx
Comment (lyudad) [inline src/components/common/wrappers/footer-wrapper.tsx:1]: move file to footer folder

File: src/components/common/wrappers/header-wrapper.tsx
Comment (lyudad) [inline src/components/common/wrappers/header-wrapper.tsx:1]: move to header folder

File: src/components/Footer/social-icons/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export { SocialIcon } from "./social-icon.tsx";
```
Comment (lyudad) [inline src/components/Footer/social-icons/index.ts]: do you need this export? from what I see it used only in SocialIcons component
in this case better to rename social-icons.tsx => index.tsx, and delete this file

File: src/components/link/index.tsx
Comment (VladManukian) [reply src/components/link/index.tsx:1]: Header should be in lower case, but it was from upper from first commint with header.
I think i rename header git mv, and everything okey. But apparently Git didn't recognize my changes to the registry. Now I'll try to fix it differently.

**Review by lyudad — APPROVED** [review]

**Pull request merged**